pytest_watch/watcher.py

```diff
@@ -164,8 +173,11 @@ def watch(directories=[], ignore=[], auto_clear=False, beep_on_failure=True,
         observer.join()
     except KeyboardInterrupt:
         observer.stop()
-    if onexit:
-        os.system(onexit)
+        if onsigint:
+            os.system(onsigint)
+    else:
+        if onexit:
+            os.system(onexit)
```

Comment:
@joeyespo 

> Why couldn't you have used `onexit` before?

The use case here is to be called when the process is interrupted (and then kill its children / running py.test).
I see that this could be handled better in general, but was the way to go when seeing the existing handlers.